addons/account/models/sequence_mixin.py

```diff
@@ -24,10 +24,22 @@ class SequenceMixin(models.AbstractModel):
     _sequence_field = "name"
     _sequence_date_field = "date"
     _sequence_index = False
-    _sequence_year_range_regex = r'^(?:(?P<prefix1>.*?)(?P<year>((?<=\D)|(?<=^))((19|20|21)\d{2}|(\d{2}(?=\D))))(?P<prefix2>\D)(?P<year_end>((?<=\D)|(?<=^))((19|20|21)\d{2}|(\d{2}(?=\D))))(?P<prefix3>\D+?))?(?P<seq>\d*)(?P<suffix>\D*?)$'
-    _sequence_monthly_regex = r'^(?P<prefix1>.*?)(?P<year>((?<=\D)|(?<=^))((19|20|21)\d{2}|(\d{2}(?=\D))))(?P<prefix2>\D*?)(?P<month>(0[1-9]|1[0-2]))(?P<prefix3>\D+?)(?P<seq>\d*)(?P<suffix>\D*?)$'
-    _sequence_yearly_regex = r'^(?P<prefix1>.*?)(?P<year>((?<=\D)|(?<=^))((19|20|21)?\d{2}))(?P<prefix2>\D+?)(?P<seq>\d*)(?P<suffix>\D*?)$'
-    _sequence_fixed_regex = r'^(?P<prefix1>.*?)(?P<seq>\d{0,9})(?P<suffix>\D*?)$'
+
+    prefix = r'(?P<prefix1>.*?)'
+    prefix2 = r'(?P<prefix2>\D)'
+    prefix3 = r'(?P<prefix3>\D+?)'
+    seq = r'(?P<seq>\d*)'
```

Comment:
Exactly, but as I did not want to go YOLO here and did not want to modify existing code for no good reason (as I don't know all context of those lines, and since the commit explicitly only modified one of the `seq` occurrences and not the others), I kept this commit as a kind of no diff commit, it only moves stuff around but does not change anything.

It's why I left a commit about this exact line in the commit msg:
```
- The `seq` part is the same everywhere except for the `fixed` case.
  It was introduced with commit [2] to not catch too long sequence to
  avoid integer too long error in postresql.
  It's ensure why it was not done in other regexes as they can still
  lead to the same error, on top of not catching the sequence the same
  way. Considering format like:
  1. INV/123456789123456789
  2. INV/2021/123456789123456789
  Why would the 1. case (fixed regex) consider the sequence as
  "123456789" (9 first digits) while case 2. (yearly regex) would
  consider the sequence as "123456789123456789" (all digits)

[1]: https://github.com/odoo/odoo/commit/1a67344694ac648af0e03df9ed6b93338a20e5b7
[2]: https://github.com/odoo/odoo/commit/49afd2f80784302ad0996741ed2c4ab57d3ac54b
```

I'll let you tell me if you really want this (not really related)) functional / business change to be part of this PR :+1: 
If so, I'll do that in a separate commit for clarity.

Reply:
Ideally it should be done as a bugfix in stable :+1:

Reply:
I'll open a PR for that and ping you, perfect :+1: